src/s390_ecc.c

```diff
@@ -616,11 +704,19 @@ unsigned int ecdsa_sign_hw(ica_adapter_handle_t adapter_handle,
 	struct ica_xcRB xcrb;
 	ECDSA_SIGN_REPLY* reply_p;
 	unsigned int privlen = privlen_from_nid(privkey->nid);
+	unsigned char X[MAX_ECC_PRIV_SIZE];
+	unsigned char Y[MAX_ECC_PRIV_SIZE];
 
 	if (adapter_handle == DRIVER_NOT_LOADED)
 		return EFAULT;
 
-	reply_p = make_ecdsa_sign_request(privkey, hash, hash_length, &xcrb);
+	rc = provide_pubkey(privkey, (unsigned char*)&X, (unsigned char*)&Y);
```

Comment:
are you sure this is (unsigned char*) &X ?
Don't you want to write
  rc = provide_pubkey(privkey, X, Y);
instead ?